src/Microsoft.AspNet.Razor/CodeGenerators/CodeWriter.cs

```diff
@@ -18,6 +19,8 @@ public class CodeWriter : IDisposable
         private int _currentLineIndex;
         private int _currentLineCharacterIndex;
 
+        public StringBuilder Builder => _writer.GetStringBuilder();
```

Comment:
1. Could `CSharpLineMappingWriter` use the existing `LastWrite` property instead? (Glad you've moved away from the weird writer -> `string` implicit conversion.)
2. If not, does the new property need to be in `CodeWriter`? `CSharpLineMappingWriter` knows it has a `CSharpCodeWriter`.